cibyl/sources/elasticsearch/api.py

```diff
@@ -38,29 +38,39 @@
 class ElasticSearch(ServerSource):
     """Elasticsearch Source"""
 
-    def __init__(self, driver: str = 'elasticsearch',
+    def __init__(self: object, driver: str = 'elasticsearch',
                  name: str = "elasticsearch", priority: int = 0,
                  elastic_client: object = None,
                  enabled: bool = True, url: str = None) -> None:
         super().__init__(name=name, driver=driver, priority=priority,
                          enabled=enabled)
         self.url = url
         self.es_client = elastic_client
+        try:
+            url_parsed = urlsplit(self.url)
+            self.host = f"{url_parsed.scheme}://{url_parsed.hostname}"
+            self.port = url_parsed.port
+        except Exception as exception:
+            raise ElasticSearchError(
+                'The URL given is not valid'
+            ) from exception
 
-    def setup(self):
+    def setup(self: object) -> None:
```

Comment:
I don't think annotating self is necessary, as far as I know the type of self should always be guaranteed to the class, since the methods are bound to the class. See also the comment in the mypy docs: https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html#classes
Relevant excerpt:
>For instance methods, omit type for "self"

Reply:
Let's discuss it in the meeting then because there's already a lot of methods with this definition.

Reply:
I deleted the type hints to those self objects.